typespec-extension/src/operation-utils.ts

```diff
@@ -51,18 +61,45 @@ export function isKnownContentType(contentTypes: string[]): boolean {
     });
 }
 
-export async function loadExamples(program: Program, options: EmitterOptions): Promise<Map<Operation, any>> {
+/**
+ * Load examples from the examples directory.
+ *
+ * @param program the program.
+ * @param options the emitter options.
+ * @param sdkContextApiVersion the apiVersion from SdkContext, for projection. It could contain "all" or "latest".
+ * @returns the Map of Operation to JSON. The Operation would be operation.projectionSource if available.
+ */
+export async function loadExamples(
+  program: Program,
+  options: EmitterOptions,
+  sdkContextApiVersion?: string,
+): Promise<Map<Operation, any>> {
   const operationExamplesMap = new Map<Operation, any>();
   const operationExamplesDirectory = options["examples-directory"];
   if (operationExamplesDirectory) {
     const operationIdExamplesMap = new Map<string, any>();
 
-    const service = ignoreDiagnostics(getAllHttpServices(program))[0];
+    let service = ignoreDiagnostics(getAllHttpServices(program))[0];
     let version = undefined;
-    const versioning = getVersion(program, service.namespace);
-    if (versioning && versioning.getVersions()) {
-      const versions = versioning.getVersions();
-      version = versions[versions.length - 1].value;
+    if (sdkContextApiVersion && !["all", "latest"].includes(sdkContextApiVersion)) {
+      version = sdkContextApiVersion;
+    } else {
+      const versioning = getVersion(program, service.namespace);
+      if (versioning && versioning.getVersions()) {
+        const versions = versioning.getVersions();
+        version = versions[versions.length - 1].value;
```

Comment:
Use `getDefaultApiVersion`? They are basically the same, just to align with https://github.com/Azure/autorest.java/blob/5a6afa35f3eeec7fecaa27fd4eb73f551788fafe/typespec-extension/src/code-model-builder.ts#L562-L569

Reply:
done